### PR TITLE
Prepare for 10.0.0 release.

### PR DESCRIPTION
**Pull request by nikhilnanivadekar**

### CODE REVIEW

File: RELEASE_NOTE_DRAFT.md [hunk 1]
```diff
@@ -1,37 +1,108 @@
...
+* Fixed several inspection violations, checkstyle violations and static analysis violations to increase code hygiene. 
+* Fixed incorrect org.eclipse.collections.api.list api doc.
+* Fixed Javadoc lint errors, legacy Javadoc to update references to Eclipse Collections.
+* Moved implementations for forEach(), toSortedListBy(), toSortedSetBy(), toSortedBagsBy(), toSortedMapBy(), toArray(), countBy(), countByWith(), selectUnique(), flatCollectWith(), reverseForEach(), reverseForEachWithIndex() as default methods to interfaces.
+* Moved implementations for tap(), select(), selectWith(), reject(), rejectWith(), collect(), collectWith(), collectIf(), flatCollect(), toImmutable(), toReversed(), reverseThis(), shuffleThis() to MutableList as default implementations.
+* Removed default implementation for RichIterable.toSortedMapBy(), MutableListFactory.ofInitialCapacity(), selectUnique() from all Bag interfaces viz. Bag, ImmutableBag, ImmutableBagIterable, ImmutablePrimitiveBag, ImmutableSortedBag, MutableBag, MutableBagIterable, MutablePrimitiveBag, MutableSortedBag, PrimitiveBag, SortedBag, UnsortedBag.
```
Comment (c-guntur) [inline RELEASE_NOTE_DRAFT.md]: duplicated in Breaking changes as well, which makes sense, but just highlighting

File: RELEASE_NOTE_DRAFT.md [hunk 2]
```diff
@@ -1,37 +1,108 @@
...
-
-Backwards Breaking Changes
+* Changed collection factories to be services that get loaded by the ServiceLoader in the API package.
+* Changed RichIterable.groupByUniqueKey() so that it's target extends MutableMapIterable instead of MutableMap.
+* Implemented UnmodifiableMutableOrderedMap.
+* Added <primitive1><primitive2>To<primitive1>Function.
+* Added specialized MultiReader interfaces for List, Set and Bag to improve the interface hierarchy.
+* Implemented RichIterable.getAny().
+* Implemented RichIterable.countByEach().
+* Implemented RichIterable.toMap().
```
Comment (c-guntur) [inline RELEASE_NOTE_DRAFT.md]: with target?

File: RELEASE_NOTE_DRAFT.md [hunk 3]
```diff
@@ -1,37 +1,108 @@
...
+* Fixed Travis builds to cache the maven local cache without including Eclipse Collections jars.
+* Removed EclipseCollectionsCodeGenerator Task.
+* Removed unused dependencies from maven build.
+* Removed obsolete maven prerequisites.
+* Customized IntelliJ maven argline settings.
+* Turned on Eclipse Collections Code Generator plugin on configuration for Eclipse IDE.
+
+# Breaking Changes
+------------------
+_Warning_: These changes are already mentioned above. The list below might not be exhaustive, make sure to test your application and usages to verify.
```
Comment (c-guntur) [inline RELEASE_NOTE_DRAFT.md]: Skipped seeing this earlier. Please ignore earlier comment about duplicates.

File: RELEASE_NOTE_DRAFT.md [hunk 4]
```diff
@@ -1,37 +1,108 @@
...
+* Changed groupByUniqueKey() so that it's target extends MutableMapIterable instead of MutableMap. This breaks binary compatibility.
+* Added missing bounded wildcards on generic types.
+* Removed deprecated classes marked for removal: SummaryStatistics, SerializableIntSummaryStatistics, SerializableLongSummaryStatistics, and SerializableDoubleSummaryStatistics.
+* Removed default implementation for RichIterable.toSortedMapBy().
+* Removed default implementation for selectUnique() from all Bag interfaces viz. Bag, ImmutableBag, ImmutableBagIterable, ImmutablePrimitiveBag, ImmutableSortedBag, MutableBag, MutableBagIterable, MutablePrimitiveBag, MutableSortedBag, PrimitiveBag, SortedBag, UnsortedBag.
+* Removed default implementation for MutableListFactory.ofInitialCapacity().
```
Comment (c-guntur) [inline RELEASE_NOTE_DRAFT.md]: Removed redundant default impl .... existed only for backward compatibility

File: RELEASE_NOTE_DRAFT.md (hunk 1 above)
Comment (nikhilnanivadekar) [reply RELEASE_NOTE_DRAFT.md]: This is expected, as mentioned in the heading Breaking Changes.

File: RELEASE_NOTE_DRAFT.md [hunk 5]
```diff
@@ -1,37 +1,106 @@
...
-* 
-
-Backwards Breaking Changes
+* Changed collection factories to be services that get loaded by the ServiceLoader in the API package.
+* Changed RichIterable.groupByUniqueKey() so that it's target extends MutableMapIterable instead of MutableMap.
+* Implemented UnmodifiableMutableOrderedMap.
+* Added <primitive1><primitive2>To<primitive1>Function.
+* Added specialized MultiReader interfaces for List, Set and Bag to improve the interface hierarchy.
+* Implemented RichIterable.getAny().
+* Implemented RichIterable.countByEach().
```
Comment (donraab) [inline RELEASE_NOTE_DRAFT.md]: countByEach is implemented as default methods on RichIterable so should not be a breaking change

File: RELEASE_NOTE_DRAFT.md (hunk 2 above)
Comment (donraab) [reply RELEASE_NOTE_DRAFT.md]: Agree with @c-guntur it should be RichIterable.toMap() with target

File: RELEASE_NOTE_DRAFT.md [hunk 6]
```diff
@@ -1,37 +1,106 @@
...
-
-* 
-
-Backwards Breaking Changes
+* Changed collection factories to be services that get loaded by the ServiceLoader in the API package.
+* Changed RichIterable.groupByUniqueKey() so that it's target extends MutableMapIterable instead of MutableMap.
+* Implemented UnmodifiableMutableOrderedMap.
+* Added <primitive1><primitive2>To<primitive1>Function.
+* Added specialized MultiReader interfaces for List, Set and Bag to improve the interface hierarchy.
+* Implemented RichIterable.getAny().
```
Comment (donraab) [inline RELEASE_NOTE_DRAFT.md]: The getAny method is also implemented as a default method so should not be a breaking change

File: RELEASE_NOTE_DRAFT.md [hunk 7]
```diff
@@ -1,37 +1,106 @@
...
+# New Functionality
 -----------------
-
-* 
-
-Backwards Breaking Changes
+* Changed collection factories to be services that get loaded by the ServiceLoader in the API package.
+* Changed RichIterable.groupByUniqueKey() so that it's target extends MutableMapIterable instead of MutableMap.
+* Implemented UnmodifiableMutableOrderedMap.
+* Added <primitive1><primitive2>To<primitive1>Function.
```
Comment (donraab) [inline RELEASE_NOTE_DRAFT.md]: The text for this does not get displayed correctly with the angle brackets.

File: RELEASE_NOTE_DRAFT.md (hunk 2 above)
Comment (nikhilnanivadekar) [reply RELEASE_NOTE_DRAFT.md]: fixed

File: RELEASE_NOTE_DRAFT.md (hunk 4 above)
Comment (nikhilnanivadekar) [reply RELEASE_NOTE_DRAFT.md]: fixed

File: RELEASE_NOTE_DRAFT.md (hunk 7 above)
Comment (nikhilnanivadekar) [reply RELEASE_NOTE_DRAFT.md]: fixed

**Pull request merged**